kombu/transport/redis_cluster.py

```diff
@@ -147,12 +147,23 @@ def _register(self, channel, client, conn, cmd):
             self._unregister(*ident)
 
         if not conn.client:
-            if conn.key in channel.ask_errors:
-                ask_error = channel.ask_errors[conn.key]
-                node = channel.client.nodes_manager.get_node(ask_error.host, ask_error.port)
-            else:
-                node = channel.client.nodes_manager.get_node_from_slot(channel.client.keyslot(conn.key))
-            conn.client = node.redis_connection.client()
+            tries = 0
+            while True:
+                if tries > 3:
+                    raise ValueError('Cannot find node for key: {}'.format(conn.key))
+                if conn.key in channel.ask_errors:
+                    ask_error = channel.ask_errors[conn.key]
+                    node = channel.client.get_node(ask_error.host, ask_error.port)
+                else:
+                    node = channel.client.get_node_from_key(conn.key)
+                if node:
+                    break
+
+                channel.client.nodes_manager.initialize()
+                tries += 1
+
+            redis_connection = channel.client.get_redis_connection(node)
```

Comment:
I can't see any differences using `get_redis_connection` and just access `redis_connection` of node since `node_manager.initialize` make all of its nodes have `redis_connection`

and wouldn't it be better to call `node_manager.initialize` before getting node rather than after getting node?

Reply:
and if i don't think it is best to call `node_manager.initialize` every making connection step

Reply:
We don't call `node_manager.initialize` if we find node correctly, so `get_redis_connection` makes a connection if we don't have a connection

Reply:
We need `node_manager.initialize` to fill in missing node, but we shouldn't call initialize every time because it resets node manager internals